subsys/bluetooth/host/att.c

```diff
@@ -3627,3 +3611,24 @@ bool bt_att_tx_meta_data_match(const struct net_buf *buf, bt_conn_tx_cb_t func,
 	return ((bt_att_tx_meta_data(buf)->func == func) &&
 		(bt_att_tx_meta_data(buf)->user_data == user_data));
 }
+
+static bool att_is_meta_data(void *user_data)
+{
+	for (struct bt_att_tx_meta_data *ptr = tx_meta_data;
+	     ptr < tx_meta_data + ARRAY_SIZE(tx_meta_data); ptr++) {
+		if ((void *)ptr == user_data) {
+			return true;
+		}
+	}
+
+	return false;
+}
```

Comment:
Could `PART_OF_ARRAY` or a simple `ptrdiff_t` expression be used here instead of a loop when you are just comparing pointer values?

Reply:
Applies to the other "is_meta_data" functions as well

Reply:
`PART_OF_ARRAY` should work. I wanted to avoid using `<`, as I believe comparing pointers from different arrays is unspecified. But if it is already in use in the tree I assume it is OK.

Reply:
> `PART_OF_ARRAY` should work. I wanted to avoid using `<`, as I believe comparing pointers from different arrays is unspecified. But if it is already in use in the tree I assume it is OK.

I guess it doesn't really make any difference if you are using `<` or `==` when comparing with different arrays :)